tracer/src/Datadog.Trace/Agent/Transports/HttpStreamRequest.cs

```diff
@@ -63,10 +81,9 @@ private async Task<IApiResponse> PostSegmentAsync(ArraySegment<byte> segment)
             using (var bidirectionalStream = _streamFactory.GetBidirectionalStream())
             {
                 var content = new BufferContent(segment);
-                var request = new HttpRequest("POST", _uri.Host, _uri.PathAndQuery, _headers, content);
-
+                _request = new HttpRequest("POST", _uri.Host, _uri.PathAndQuery, _headers, content);
                 // send request, get response
-                var response = await _client.SendAsync(request, bidirectionalStream, bidirectionalStream).ConfigureAwait(false);
+                var response = await _client.SendAsync(_request, bidirectionalStream, bidirectionalStream).ConfigureAwait(false);
```

Comment:
Thanks for the update @anna-git 🙂 This mutable `_request` variable still bothers me a bit though, as it's only non-null/valid to access at certain points. Could we change the signature of the private `PostSegmentAsync` method to return _both_ the `IApiResponse` and the `HttpRequest`? That would remove  the need for the field

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/1778/commits/62c946f8890c5547eb463e90479a50f783e369bf